### DISCUSSION TITLE
Discussion: parallel execution during nox sessions

### DISCUSSION THREAD

**Discussion by staticdev** [cjolowicz/cookiecutter-hypermodern-python#773] [Ideas]
I am aware that `nox` does not have built-in parallel capabilities like `tox -p` has. There is an old open issue for that https://github.com/theacodes/nox/issues/88

But maybe we can do something to have more CPU usage during the execution an bring the time down. Adding [pytest-xdist](https://github.com/pytest-dev/pytest-xdist) on test runs for example? Or maybe some other trick you are aware of.

**Comment by cjolowicz** [comment]
I'm hesitant about that because for all I know, it's tricky to set this up correctly with coverage. If you can figure this out, I'd be happy to take a look. Apparently this is one of the reasons that `pytest-cov` exists. If you go down this route, I'd like to keep using the Coverage.py interface as much as possible, rather than configuring Coverage through the pytest-cov CLI.

**Reply by AdrianCert** [reply]
We enhance this by store into different files (1 .coverage per process/session/thread) and combine those later on!
